ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -154,11 +159,19 @@ public KsqlResource(
     this.registerKsqlStatementTasks();
   }
 
+  // CHECKSTYLE_RULES.OFF: NPathComplexity
```

Comment:
We can avoid disabling this check by merging the new try-catch block with the existing one for `ksqlEngine.parseStatements`, but my gut says they should be separate. Thoughts?

Reply:
I think a better approach would be to do the error code handling from waitForCommandOffset.

Reply:
I've a big PR which will address the complexity. 

The race is on to see who merges first!